Add documentation for wait_for_completion API (#3458)
opensearch-project/documentation-website: _api-reference/index-apis/clone.md

wait_for_active_shards | String | The number of active shards that must be available before OpenSearch processes the request. Default is 1 (only the primary shard). Set to all or a positive integer. Values greater than 1 require replicas. For example, if you specify a value of 3, the index must have two replicas distributed across two additional nodes for the operation to succeed.
master_timeout | Time | How long to wait for a connection to the master node. Default is `30s`.
timeout | Time | How long to wait for the request to return. Default is `30s`.

> wait_for_completion | Boolean | When set to `false`, the request will return a task immediately and the operation will run in background. Default is `true`.

Reviewer: I assume it's the task **ID** that's returned? And then you use that task ID in the tasks API to check on the task? If so, add this information and link to the tasks API.
```suggestion
wait_for_completion | Boolean | When set to `false`, the request will return immediately instead of waiting for the operation to finish. Default is `true`.
```

Author: yes, a task ID is returned. I'll get this updated.

> wait_for_completion | Boolean | When set to `false`, the request will return a [task]({{site.url}}{{site.baseurl}}/api-reference/tasks/) ID and the operation will run in background. Default is `true`.

Reviewer: Because the operation runs on the server, I'm not sure "in the background" is the correct verbiage here. I would go with 
```suggestion
wait_for_completion | Boolean | When set to `false`, the request will return immediately instead of waiting for the operation to finish. To monitor the operation status, use the [Tasks API]({{site.url}}{{site.baseurl}}/api-reference/tasks/) with the task ID that the request returns. Default is `true`.
```

Author: ok, I'll get it updated

> wait_for_completion | Boolean | When set to `false`, the request will return immediately instead of waiting for the operation to finish. To monitor the operation status, use the [Tasks API]({{site.url}}{{site.baseurl}}/api-reference/tasks/) with the task ID that the request returns. Default is `true`.

Reviewer 2: ```suggestion
wait_for_completion | Boolean | When set to `false`, the request will be returned immediately instead of after the operation is finished. To monitor the operation status, use the [Tasks API]({{site.url}}{{site.baseurl}}/api-reference/tasks/) with the task ID returned by the request. Default is `true`.
```

> task_execution_timeout | Time | Explicit task execution timeout, only useful when wait_for_completion is false. Default is `1h`.

Reviewer 2: ```suggestion
task_execution_timeout | Time | The explicit task execution timeout. Only useful when wait_for_completion is set to `false`. Default is `1h`.
```
